DMCompiler/Compiler/DMPreprocessor/DMPreprocessor.cs

```diff
@@ -541,6 +575,76 @@ private List<Token> GetLineOfTokens()
             }
         }
 
+        private void HandlePragmaDirective(Token pragmaDirective) {
+            Token warningNameToken = GetNextToken(true);
+            WarningCode warningCode;
+            switch(warningNameToken.Type) {
+                case TokenType.DM_Preproc_Identifier: {
+                    if(!Enum.TryParse<WarningCode>(warningNameToken.Text, out warningCode)) {
+                        DMCompiler.Emit(WarningCode.BadDirective, warningNameToken.Location, $"Warning '{warningNameToken.PrintableText}' does not exist");
+                        GetLineOfTokens(); // consume what's on this line and leave
+                        return;
+                    }
+                    break;
+                }
+                case TokenType.DM_Preproc_Number: {
+                    if(!Int32.TryParse(warningNameToken.Text, out var intValue)) {
+                        DMCompiler.Emit(WarningCode.BadDirective, warningNameToken.Location, $"Warning OD{warningNameToken.PrintableText} does not exist");
+                        GetLineOfTokens();
+                        return;
+                    }
+                    warningCode = (WarningCode)intValue;
+                    break;
+                }
+                default: {
+                    DMCompiler.Emit(WarningCode.BadDirective, warningNameToken.Location, $"Invalid warning identifier '{warningNameToken.PrintableText}'");
+                    GetLineOfTokens();
+                    return;
+                }
+            }
+            DebugTools.AssertNotNull(warningCode);
+            if((int)warningCode < 1000) {
+                DMCompiler.Emit(WarningCode.BadDirective, warningNameToken.Location, $"Warning OD{(int)warningCode:d4} cannot be set - it must always be an error");
+                GetLineOfTokens();
+                return;
+            }
+
+            Token warningTypeToken = GetNextToken(true);
+            if (warningTypeToken.Type != TokenType.DM_Preproc_Identifier) {
+                DMCompiler.Emit(WarningCode.BadDirective, warningNameToken.Location, $"Warnings can only be set to disabled, notice, warning, or error");
+                return;
+            }
+            switch(warningTypeToken.Text.ToLower()) {
+                case "disabled":
+                case "disable":
+                    DMCompiler.SetPragma(warningCode, ErrorLevel.Disabled);
+                    break;
+                case "notice":
```

Comment:
```suggestion
                case "notice":
                case "info":
```
pretty sure I've seen this somewhere